Zend/zend_operators.c

```diff
@@ -1655,7 +1698,19 @@ ZEND_API int ZEND_FASTCALL concat_function(zval *result, zval *op1, zval *op2) /
 
 		if (UNEXPECTED(op1_len > SIZE_MAX - op2_len)) {
 			zend_throw_error(NULL, "String size overflow");
-			ZVAL_FALSE(result);
+
+			if (UNEXPECTED(use_copy1)) {
+				zval_dtor(op1);
+				if (Z_ISUNDEF(orig_op1)) {
+					ZVAL_UNDEF(result);
+				}
+			} else if (result == op1) {
```

Comment:
Should this be `result != op1`?

TBH this is pretty ugly -- given that it's an existing issue, I'd be fine with just leaving it be.